test/groovy/PiperExecuteBinTest.groovy

```diff
@@ -209,6 +213,50 @@ class PiperExecuteBinTest extends BasePiperTest {
         assertThat(sshKey, is(['sshKey']))
     }
 
+    @Test
+    void testPiperExecuteBinCredentialsSetAsEnum() {
```

Comment:
Well, I'm not an expert, so I apologize in advance for comments which might be weird. 
What is the intention of this test, feed all credentials of the enum into piperExecuteBin() and check the outcome? Shouldn't the test run the credentials separately, one-by-one instead of all-at-once?
  testFile
  testToken
  test...
The reader could image the method need all credentials.

Isn't testPiperExecuteBinSSHCredentials redundant?

Reply:
>What is the intention of this test, ...

The idea is for the test to fail should somebody introduce a new credential type to the switch in the piperExecuteBin step, without adding it to the enum as well.

Reply:
>Isn't testPiperExecuteBinSSHCredentials redundant?

Yes, it was removed now.